zkevm-circuits/src/evm_circuit.rs

```diff
@@ -16,37 +16,32 @@ use execution::ExecutionConfig;
 use table::{FixedTableTag, LookupTable};
 use witness::Block;
 
-use self::param::STEP_HEIGHT;
-
 /// EvmCircuit implements verification of execution trace of a block.
 #[derive(Clone, Debug)]
 pub struct EvmCircuit<F> {
     fixed_table: [Column<Fixed>; 4],
+    byte_table: [Column<Fixed>; 1],
     execution: ExecutionConfig<F>,
 }
 
 impl<F: Field> EvmCircuit<F> {
     /// Configure EvmCircuit
-    pub fn configure<TxTable, RwTable, BytecodeTable, BlockTable>(
+    pub fn configure(
         meta: &mut ConstraintSystem<F>,
         power_of_randomness: [Expression<F>; 31],
-        tx_table: TxTable,
-        rw_table: RwTable,
-        bytecode_table: BytecodeTable,
-        block_table: BlockTable,
-    ) -> Self
-    where
-        TxTable: LookupTable<F, 4>,
-        RwTable: LookupTable<F, 11>,
-        BytecodeTable: LookupTable<F, 4>,
-        BlockTable: LookupTable<F, 3>,
-    {
+        tx_table: &dyn LookupTable<F>,
+        rw_table: &dyn LookupTable<F>,
+        bytecode_table: &dyn LookupTable<F>,
+        block_table: &dyn LookupTable<F>,
```

Comment:
Oh wasn't aware this was something that could be dangerous! I'm still pretty new to rust so don't always know what the best practices are. If it's something we want to avoid will certainly remove. Will also look into this some more so I better know the potential pitfalls. Thanks for the warning!

Reply:
Nonono!! It's not dangerous. And it's meant to be used as `&impl LookupTable` for example.

But using `dynamic dispatch` brings in dyn-safety concerns when using generics. It's fine as this is the correct solution. But I just raise the flag as dyn-safety rules are pretty restrictive. So we will need to play with references and explicit lifetimes more.

All I mean is that `LookupTable` trait will not be able to have certain methods or logic. And we need to count on it from now on :)

See: https://doc.rust-lang.org/nightly/reference/items/traits.html#object-safety

See also a really interesting article (in case you are curious and like Rust) where `dyn*` is discussed and the caveats tha `& dyn` currently has: https://smallcultfollowing.com/babysteps/blog/2022/03/29/dyn-can-we-make-dyn-sized/